src/hypofuzz/dashboard/dashboard.py

```diff
@@ -559,6 +547,37 @@ def send_nowait_from_anywhere(msg: ListenerEventT) -> None:
         )
         TESTS[fuzz_target.nodeid] = test
 
+        async def update_websockets() -> None:
+            # TODO: make this more granular, so we send incremental batches
+            # of reports as they're loaded, etc.
+            for websocket in websockets.copy():
+                await websocket.initial({test.nodeid: test})
+
+        trio.from_thread.run(update_websockets, trio_token=trio_token)
+
+
+async def run_dashboard(port: int, host: str) -> None:
+    send_channel, receive_channel = trio.open_memory_channel[ListenerEventT](math.inf)
+    trio_token = trio.lowlevel.current_trio_token()
+
+    def send_nowait_from_anywhere(msg: ListenerEventT) -> None:
+        # DirectoryBasedExampleDatabase sends events from a background thread (via watchdog),
+        # so we need to support sending from anywhere, i.e. whether or not the calling thread
+        # has any Trio state.  We can do that with the following branch:
+        try:
+            trio.lowlevel.current_task()
+        except RuntimeError:
+            trio.from_thread.run_sync(
+                send_channel.send_nowait, msg, trio_token=trio_token
+            )
+        else:
+            send_channel.send_nowait(msg)
+
+    db = get_db()
+    thread = threading.Thread(
+        target=_load_initial_state, kwargs={"trio_token": trio_token}
+    )
+    thread.start()
     # Any database events that get submitted while we're computing initial state
     # won't be displayed until a dashboard restart. We could solve this by adding the
     # listener first, then storing reports in a queue, to be resolved after
```

Comment:
We actually can't solve this in full generality (we don't have a globally consistent view), but we _can_ ensure that the events lost are only while we're loading state for _this particular_ test.  

Basically: start the listener _before_ loading initial state, but only keep reported changes from after that test finishes loading.  Any changes from before the test starts loading will be picked up by the initial load anyway, so it's only an issue for cases where something is written-but-not-read _during_ the load.